### ISSUE TITLE
Linting exception of GH Actions when \t at EOL

### ISSUE THREAD

**Issue by jpfeuffer** [nf-core/tools#790]
## Description of the bug

When you have a tab at the end of a line in a GH Actions workflow yml, you get an exception.
https://github.com/nf-core/proteomicslfq/pull/114/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR20

Exception:
```
INFO     Testing pipeline:                                           lint.py:201
         /home/runner/work/proteomicslfq/proteomicslfq                          
╭───────────────────── Traceback (most recent call last) ──────────────────────╮
│ /opt/hostedtoolcache/Python/3.6.12/x64/bin/nf-core:8 in <module>             │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/nf_core/_ │
│ _main__.py:56 in run_nf_core                                                 │
│                                                                              │
│    53 │   stderr.print("\n\n")                                               │
│    54 │                                                                      │
│    55 │   # Lanch the click cli                                              │
│ ❱  56 │   nf_core_cli()                                                      │
│    57                                                                        │
│    58                                                                        │
│    59 # Customise the order of subcommands for --help                        │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/click/cor │
│ e.py:829 in __call__                                                         │
│                                                                              │
│    826 │                                                                     │
│    827 │   def __call__(self, *args, **kwargs):                              │
│    828 │   │   """Alias for :meth:`main`."""                                 │
│ ❱  829 │   │   return self.main(*args, **kwargs)                             │
│    830                                                                       │
│    831                                                                       │
│    832 class Command(BaseCommand):                                           │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/click/cor │
│ e.py:782 in main                                                             │
│                                                                              │
│    779 │   │   try:                                                          │
│    780 │   │   │   try:                                                      │
│    781 │   │   │   │   with self.make_context(prog_name, args, **extra) as c │
│ ❱  782 │   │   │   │   │   rv = self.invoke(ctx)                             │
│    783 │   │   │   │   │   if not standalone_mode:                           │
│    784 │   │   │   │   │   │   return rv                                     │
│    785 │   │   │   │   │   # it's not safe to `ctx.exit(rv)` here!           │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/click/cor │
│ e.py:1259 in invoke                                                          │
│                                                                              │
│   1256 │   │   │   │   Command.invoke(self, ctx)                             │
│   1257 │   │   │   │   sub_ctx = cmd.make_context(cmd_name, args, parent=ctx │
│   1258 │   │   │   │   with sub_ctx:                                         │
│ ❱ 1259 │   │   │   │   │   return _process_result(sub_ctx.command.invoke(sub │
│   1260 │   │                                                                 │
│   1261 │   │   # In chain mode we create the contexts step by step, but afte │
│   1262 │   │   # base command has been invoked.  Because at that point we do │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/click/cor │
│ e.py:1066 in invoke                                                          │
│                                                                              │
│   1063 │   │   """                                                           │
│   1064 │   │   _maybe_show_deprecated_notice(self)                           │
│   1065 │   │   if self.callback is not None:                                 │
│ ❱ 1066 │   │   │   return ctx.invoke(self.callback, **ctx.params)            │
│   1067                                                                       │
│   1068                                                                       │
│   1069 class MultiCommand(Command):                                          │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/click/cor │
│ e.py:610 in invoke                                                           │
│                                                                              │
│    607 │   │   args = args[2:]                                               │
│    608 │   │   with augment_usage_errors(self):                              │
│    609 │   │   │   with ctx:                                                 │
│ ❱  610 │   │   │   │   return callback(*args, **kwargs)                      │
│    611 │                                                                     │
│    612 │   def forward(*args, **kwargs):  # noqa: B902                       │
│    613 │   │   """Similar to :meth:`invoke` but fills in default keyword     │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/nf_core/_ │
│ _main__.py:306 in lint                                                       │
│                                                                              │
│   303 │   """                                                                │
│   304 │                                                                      │
│   305 │   # Run the lint tests!                                              │
│ ❱ 306 │   lint_obj = nf_core.lint.run_linting(pipeline_dir, release, show_pa │
│       json)                                                                  │
│   307 │   if len(lint_obj.failed) > 0:                                       │
│   308 │   │   sys.exit(1)                                                    │
│   309                                                                        │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/nf_core/l │
│ int.py:61 in run_linting                                                     │
│                                                                              │
│     58 │                                                                     │
│     59 │   # Run the linting tests                                           │
│     60 │   try:                                                              │
│ ❱   61 │   │   lint_obj.lint_pipeline(release_mode)                          │
│     62 │   except AssertionError as e:                                       │
│     63 │   │   log.critical("Critical error: {}".format(e))                  │
│     64 │   │   log.info("Stopping tests...")                                 │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/nf_core/l │
│ int.py:240 in lint_pipeline                                                  │
│                                                                              │
│    237 │   │   │   for fun_name in check_functions:                          │
│    238 │   │   │   │   progress.update(lint_progress, advance=1, func_name=f │
│    239 │   │   │   │   log.debug("Running lint test: {}".format(fun_name))   │
│ ❱  240 │   │   │   │   getattr(self, fun_name)()                             │
│    241 │   │   │   │   if len(self.failed) > 0:                              │
│    242 │   │   │   │   │   log.error("Found test failures in `{}`, halting l │
│        run.".format(fun_name))                                               │
│    243 │   │   │   │   │   break                                             │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/nf_core/l │
│ int.py:698 in check_actions_ci                                               │
│                                                                              │
│    695 │   │   fn = os.path.join(self.path, ".github", "workflows", "ci.yml" │
│    696 │   │   if os.path.isfile(fn):                                        │
│    697 │   │   │   with open(fn, "r") as fh:                                 │
│ ❱  698 │   │   │   │   ciwf = yaml.safe_load(fh)                             │
│    699 │   │   │                                                             │
│    700 │   │   │   # Check that the action is turned on for the correct even │
│    701 │   │   │   try:                                                      │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/__in │
│ it__.py:162 in safe_load                                                     │
│                                                                              │
│   159 │   to be safe for untrusted input.                                    │
│   160 │   """                                                                │
│   161 │   return load(stream, SafeLoader)                                    │
│ ❱ 162                                                                        │
│   163 def safe_load_all(stream):                                             │
│   164 │   """                                                                │
│   165 │   Parse all YAML documents in a stream                               │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/__in │
│ it__.py:114 in load                                                          │
│                                                                              │
│   111 │   loader = Loader(stream)                                            │
│   112 │   try:                                                               │
│   113 │   │   return loader.get_single_data()                                │
│ ❱ 114 │   finally:                                                           │
│   115 │   │   loader.dispose()                                               │
│   116                                                                        │
│   117 def load_all(stream, Loader=None):                                     │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/cons │
│ tructor.py:49 in get_single_data                                             │
│                                                                              │
│    46 │   def get_single_data(self):                                         │
│    47 │   │   # Ensure that the stream contains a single document and constr │
│    48 │   │   node = self.get_single_node()                                  │
│ ❱  49 │   │   if node is not None:                                           │
│    50 │   │   │   return self.construct_document(node)                       │
│    51 │   │   return None                                                    │
│    52 │                                                                      │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/comp │
│ oser.py:36 in get_single_node                                                │
│                                                                              │
│    33 │   │   document = None                                                │
│    34 │   │   if not self.check_event(StreamEndEvent):                       │
│    35 │   │   │   document = self.compose_document()                         │
│ ❱  36 │   │                                                                  │
│    37 │   │   # Ensure that the stream contains no more documents.           │
│    38 │   │   if not self.check_event(StreamEndEvent):                       │
│    39 │   │   │   event = self.get_event()                                   │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/comp │
│ oser.py:55 in compose_document                                               │
│                                                                              │
│    52 │   │                                                                  │
│    53 │   │   # Compose the root node.                                       │
│    54 │   │   node = self.compose_node(None, None)                           │
│ ❱  55 │   │                                                                  │
│    56 │   │   # Drop the DOCUMENT-END event.                                 │
│    57 │   │   self.get_event()                                               │
│    58 │   │                                                                  │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/comp │
│ oser.py:84 in compose_node                                                   │
│                                                                              │
│    81 │   │   │   node = self.compose_sequence_node(anchor)                  │
│    82 │   │   elif self.check_event(MappingStartEvent):                      │
│    83 │   │   │   node = self.compose_mapping_node(anchor)                   │
│ ❱  84 │   │   self.ascend_resolver()                                         │
│    85 │   │   return node                                                    │
│    86 │                                                                      │
│    87 │   def compose_scalar_node(self, anchor):                             │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/comp │
│ oser.py:133 in compose_mapping_node                                          │
│                                                                              │
│   130 │   │   │   #    raise ComposerError("while composing a mapping",      │
│       start_event.start_mark,                                                │
│   131 │   │   │   #            "found duplicate key", key_event.start_mark)  │
│   132 │   │   │   item_value = self.compose_node(node, item_key)             │
│ ❱ 133 │   │   │   #node.value[item_key] = item_value                         │
│   134 │   │   │   node.value.append((item_key, item_value))                  │
│   135 │   │   end_event = self.get_event()                                   │
│   136 │   │   node.end_mark = end_event.end_mark                             │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/comp │
│ oser.py:84 in compose_node                                                   │
│                                                                              │
│    81 │   │   │   node = self.compose_sequence_node(anchor)                  │
│    82 │   │   elif self.check_event(MappingStartEvent):                      │
│    83 │   │   │   node = self.compose_mapping_node(anchor)                   │
│ ❱  84 │   │   self.ascend_resolver()                                         │
│    85 │   │   return node                                                    │
│    86 │                                                                      │
│    87 │   def compose_scalar_node(self, anchor):                             │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/comp │
│ oser.py:133 in compose_mapping_node                                          │
│                                                                              │
│   130 │   │   │   #    raise ComposerError("while composing a mapping",      │
│       start_event.start_mark,                                                │
│   131 │   │   │   #            "found duplicate key", key_event.start_mark)  │
│   132 │   │   │   item_value = self.compose_node(node, item_key)             │
│ ❱ 133 │   │   │   #node.value[item_key] = item_value                         │
│   134 │   │   │   node.value.append((item_key, item_value))                  │
│   135 │   │   end_event = self.get_event()                                   │
│   136 │   │   node.end_mark = end_event.end_mark                             │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/comp │
│ oser.py:84 in compose_node                                                   │
│                                                                              │
│    81 │   │   │   node = self.compose_sequence_node(anchor)                  │
│    82 │   │   elif self.check_event(MappingStartEvent):                      │
│    83 │   │   │   node = self.compose_mapping_node(anchor)                   │
│ ❱  84 │   │   self.ascend_resolver()                                         │
│    85 │   │   return node                                                    │
│    86 │                                                                      │
│    87 │   def compose_scalar_node(self, anchor):                             │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/comp │
│ oser.py:133 in compose_mapping_node                                          │
│                                                                              │
│   130 │   │   │   #    raise ComposerError("while composing a mapping",      │
│       start_event.start_mark,                                                │
│   131 │   │   │   #            "found duplicate key", key_event.start_mark)  │
│   132 │   │   │   item_value = self.compose_node(node, item_key)             │
│ ❱ 133 │   │   │   #node.value[item_key] = item_value                         │
│   134 │   │   │   node.value.append((item_key, item_value))                  │
│   135 │   │   end_event = self.get_event()                                   │
│   136 │   │   node.end_mark = end_event.end_mark                             │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/comp │
│ oser.py:84 in compose_node                                                   │
│                                                                              │
│    81 │   │   │   node = self.compose_sequence_node(anchor)                  │
│    82 │   │   elif self.check_event(MappingStartEvent):                      │
│    83 │   │   │   node = self.compose_mapping_node(anchor)                   │
│ ❱  84 │   │   self.ascend_resolver()                                         │
│    85 │   │   return node                                                    │
│    86 │                                                                      │
│    87 │   def compose_scalar_node(self, anchor):                             │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/comp │
│ oser.py:127 in compose_mapping_node                                          │
│                                                                              │
│   124 │   │   if anchor is not None:                                         │
│   125 │   │   │   self.anchors[anchor] = node                                │
│   126 │   │   while not self.check_event(MappingEndEvent):                   │
│ ❱ 127 │   │   │   #key_event = self.peek_event()                             │
│   128 │   │   │   item_key = self.compose_node(node, None)                   │
│   129 │   │   │   #if item_key in node.value:                                │
│   130 │   │   │   #    raise ComposerError("while composing a mapping",      │
│       start_event.start_mark,                                                │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/pars │
│ er.py:98 in check_event                                                      │
│                                                                              │
│    95 │   │   if self.current_event is None:                                 │
│    96 │   │   │   if self.state:                                             │
│    97 │   │   │   │   self.current_event = self.state()                      │
│ ❱  98 │   │   if self.current_event is not None:                             │
│    99 │   │   │   if not choices:                                            │
│   100 │   │   │   │   return True                                            │
│   101 │   │   │   for choice in choices:                                     │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/pars │
│ er.py:428 in parse_block_mapping_key                                         │
│                                                                              │
│   425 │                                                                      │
│   426 │   def parse_block_mapping_key(self):                                 │
│   427 │   │   if self.check_token(KeyToken):                                 │
│ ❱ 428 │   │   │   token = self.get_token()                                   │
│   429 │   │   │   if not self.check_token(KeyToken, ValueToken, BlockEndToke │
│   430 │   │   │   │   self.states.append(self.parse_block_mapping_value)     │
│   431 │   │   │   │   return self.parse_block_node_or_indentless_sequence()  │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/scan │
│ ner.py:116 in check_token                                                    │
│                                                                              │
│    113 │   │   # Check if the next token is one of the given types.          │
│    114 │   │   while self.need_more_tokens():                                │
│    115 │   │   │   self.fetch_more_tokens()                                  │
│ ❱  116 │   │   if self.tokens:                                               │
│    117 │   │   │   if not choices:                                           │
│    118 │   │   │   │   return True                                           │
│    119 │   │   │   for choice in choices:                                    │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/yaml/scan │
│ ner.py:260 in fetch_more_tokens                                              │
│                                                                              │
│    257 │   │   raise ScannerError("while scanning for the next token", None, │
│    258 │   │   │   │   "found character %r that cannot start any token" % ch │
│    259 │   │   │   │   self.get_mark())                                      │
│ ❱  260 │                                                                     │
│    261 │   # Simple keys treatment.                                          │
│    262 │                                                                     │
│    263 │   def next_possible_simple_key(self):                               │
╰──────────────────────────────────────────────────────────────────────────────╯
ScannerError: while scanning for the next token
found character '\t' that cannot start any token
  in "/home/runner/work/proteomicslfq/proteomicslfq/.github/workflows/ci.yml", 
line 20, column 67
```
## Steps to reproduce

Steps to reproduce the behaviour:

No minimal example yet, happened in https://github.com/nf-core/proteomicslfq/runs/1431301648

## Expected behaviour

Just complain about the superfluous tab.

## System

- GH actions runner

## Nextflow Installation

- any

**Comment by ewels** [comment]
Strange! Good spot.

**Comment by KevinMenden** [comment]
Could replicate this - although vscode tried really hard to prevent me from using tabs in YAML files :) 

Having a tab in YAML files is not allowed: 
https://yaml.org/spec/1.2/spec.html#id2777534

This should basically be solved with #820 , which catches errors when parsing YAML files.
Only thing I just realised is that we don't report the error.
So it might be worth catching that `ScannerError` to give the user a better idea of what is wrong with the YAML file if it can't be loaded.

**Comment by KevinMenden** [comment]
Added full exception reporting now to all the places where action yaml files are loaded in #822
(should have done that at last PR anyway :) ) 

The issue is basically already resolved, the error is catched. The new PR will only make the message nicer, so I'm closing this issue.
